lib/absinthe/relay/connection/notation.ex

```diff
@@ -210,17 +250,34 @@ defmodule Absinthe.Relay.Connection.Notation do
   @doc false
   # Record the edge object
   def record_edge_object!(env, naming, attrs, block) do
-    Notation.record_object!(env, naming.edge_type_identifier, attrs, [
-      block,
-      edge_object_body(naming, block)
-    ])
+    Notation.record_object!(
+      env,
+      naming.edge_type_identifier,
+      Keyword.delete(attrs, :non_null_node_type),
+      [
+        block,
+        edge_object_body(naming, block)
+      ]
+    )
   end
 
+  # Encode the non-null setting appropriate for storage with
+  # `Absinthe.Schema.Notation.private/3`
+  defp encode_non_null_node_type_private_value(false), do: :set_false
+  defp encode_non_null_node_type_private_value(true), do: :set_true
+
+  # Decode the non-null setting (see `encode_non_null_node_type_private_value/1`)
+  defp decode_non_null_node_type_private_value(:set_false), do: false
+  defp decode_non_null_node_type_private_value(:set_true), do: true
+
   defp connection_object_body(naming) do
     edge_type = naming.edge_type_identifier
     [private_category_node, private_key_node] = @private_node_type_identifier_path
     [private_category_base, private_key_base] = @private_base_identifier_path
 
+    [private_category_non_null_node_type, private_key_non_null_node_type] =
+      @private_non_null_node_type_path
+
     quote do
       field :page_info, type: non_null(:page_info)
       field :edges, type: list_of(unquote(edge_type))
```

Comment:
So this type would preferably also be configurable to be `non_null(list_of(...))`